frontend/fontlist.lua

```diff
@@ -95,21 +102,65 @@ local function getExternalFontDir()
     end
 end
 
-local function _readList(target, dir)
+-- Query FreeType/HarfBuzz about font metadata
+local function collectFaceInfo(path)
+    local res = {}
+    local n = FT.getFaceCount(path)
+    if not n then
+        return
+    end
+    for i=0, n-1 do
+        local ok, face = pcall(FT.newFace, path, nil, i)
+        if not ok then
+            return nil
+        end
+
+        local fres = face:getInfo()
+        local hbface = HB.hb_ft_face_create_referenced(face)
+        fres.names = hbface:getNames()
+        fres.scripts, fres.langs = hbface:getCoverage()
+        fres.path = path
+        fres.index = i
+        table.insert(res, fres)
+
+        hbface:destroy()
+        face:done()
+    end
+    return res
+end
+
+local font_exts = {
+    ["ttf"] = true,
+    ["ttc"] = true,
+    ["cff"] = true,
+    ["otf"] = true,
+}
+
+local function _readList(target, tinfo, mark, dir)
```

Comment:
Might give a better name to this `tinfo` (which is `fontinfo` in the caller, so why not `fontinfo` here too ?)

Reply:
I'm tempted to move this into a local lambda. There are now 3 variables passed in the recursive call, all of which are fixed state. Would be much better off being captured by the closure at this point.

Reply:
(split into util.findFiles in 670473d3b70d0388355da92e88617d1d40232aca, as this was badly nested already)